File: chat2db-community-server/chat2db-community-domain/chat2db-community-domain-api/src/main/java/ai/chat2db/community/domain/api/model/agent/AgentDataWikiBinding.java
package ai.chat2db.community.domain.api.model.agent;

import ai.chat2db.community.domain.api.enums.agent.AgentApprovalModeEnum;
import lombok.Data;

/**
 * Agent-specific execution policy applied to every table contributed by one DataWiki.
 */
@Data
public class AgentDataWikiBinding {

    private String dataWikiId;

    private Integer maxRows = 200;

    private Integer timeoutSeconds = 60;

    private AgentApprovalModeEnum approvalMode = AgentApprovalModeEnum.RISK_BASED;

    private Boolean allowProduction = false;
}

File: chat2db-community-client/src/pages/main/tasks/agentDataWikiBinding.ts
import type { AgentDataWikiBinding, AgentDefinition } from '@/service/agent';

export const agentDataWikiBindings = (agent?: AgentDefinition): AgentDataWikiBinding[] =>
  agent?.dataWikiBindings?.length
    ? agent.dataWikiBindings.map((binding) => ({ ...binding }))
    : (agent?.dataWikiIds || []).map((dataWikiId) => ({
        dataWikiId,
        maxRows: 200,
        timeoutSeconds: 60,
        approvalMode: 'RISK_BASED',
        allowProduction: false,
      }));

export const dataWikiBindingIds = (bindings?: AgentDataWikiBinding[]): string[] =>
  (bindings || []).map((binding) => binding.dataWikiId);
